activerecord/lib/active_record/insert_all.rb

```diff
@@ -7,11 +7,12 @@ class InsertAll # :nodoc:
     attr_reader :model, :connection, :inserts, :keys
     attr_reader :on_duplicate, :returning, :unique_by, :update_sql
 
-    def initialize(model, inserts, on_duplicate:, returning: nil, unique_by: nil)
+    def initialize(model, inserts, on_duplicate:, returning: nil, unique_by: nil, set_timestamps_on_create: false)
```

Comment:
I made this behaviour configurable, as I figured some people might want to opt-out for whatever reasons, similarly to being able to opt-out of setting `updated_at` and `updated_on` automatically on upserts.

I made it default to `false` because I wasn't sure if we'd want to default to `true` right off the bat. That said, with Rails 7 coming up, maybe we could? 🤷 

I think ideally, we'd either ship with it on by default, or set it to on by default in a subsequent version of ActiveRecord.